src/main/java/htsjdk/samtools/TextualBAMIndexWriter.java

```diff
@@ -71,14 +71,14 @@ private void writeHeader() {
     @Override
     public void writeReference(final BAMIndexContent content) {
 
-        final int reference = content.getReferenceSequence();
-
         if (content == null) {
-            writeNullContent(reference);
+            writeNullContent();
```

Comment:
This changes the behavior; before this would throw a `NullPointerException`. Is the new behavior the correct/desired behavior of this API?

Reply:
The NPE was introduced in https://github.com/samtools/htsjdk/commit/ee10696af44fcf6d0f2166b14884f7ae7b0e6aeb#diff-d029b962e0444d3ad90c9c69372591bcL70. This fix removes the NPE, which seems like the right thing to do.

Reply:
It was introduced a month after the initial checkin of the code, which was 8 years ago. So it's worked this way for the lifetime of this code. If `content == null` is not a valid state, it would be better to remove this code rather than add a new feature.

Reply:
Fair enough. I've fixed this to explicitly throw a NPE in this case.